### PR TITLE
Fix Hydration failed

### PR DESCRIPTION
**Pull request by geeknees**
Controls rendering being called at unintended times

### CODE REVIEW

**Comment by SamuraiT** [comment]
> Controls rendering being called at unintended times
aha, which component or variable ?

File: hooks/badge/useMintBadge.ts [hunk 1]
```diff
@@ -1,30 +1,29 @@
...
-import { useContractWrite, useContractEvent, useNetwork, useAccount } from 'wagmi'
+import { useContractWrite, useContractEvent, useAccount } from 'wagmi'
 import henkakuBadge from '@/utils/abis/henkakuBadge.json'
-import { getContractAddress } from '@/utils/contractAddress'
 
 export const useMintBadge = (
+  contract: string,
+  owner: string | undefined,
   tokenID: number
 ) => {
```
Comment (SamuraiT) [inline hooks/badge/useMintBadge.ts:10]: ideally, better to have contract and owner inside of hooks.
wdyt ?

Comment (geeknees) [reply hooks/badge/useMintBadge.ts:10]: That's one thing, but the problem is that we have to call useNetwork multiple times. Not only is this error-prone, but it is also a performance concern. And that is only to get the address of the contract.

If we were to change the contract address to a method of switching by environment variables, I don't think there would be a problem with putting it in the contract hooks.

**Comment by geeknees** [comment]
It is actually difficult to pinpoint the problematic areas, but the following are suspect areas.

```js
const { isConnected } = useConnect()
```
```js
                {hasNft && t('title.minted')}
                {!badge?.mintable && !hasNft && t('title.notMintable')}
                {badge?.mintable && badge?.amount.gt(0) && !hasNft && (
                {badge?.mintable && badge?.amount.eq(0) && !hasNft && (
```

**Comment by SamuraiT** [comment]
> const { isConnected } = useConnect()

maybe this one.

File: hooks/badge/useMintBadge.ts (hunk 1 above)
Comment (SamuraiT) [reply hooks/badge/useMintBadge.ts:10]: gotcha

**Review by SamuraiT — APPROVED** [review]

**Pull request merged**